File: evaluation/recpro-evaluation-framework/src/main/java/de/ubt/ai4/petter/recpro/recpro_evaluation_framework/util/api/ApiConstants.java
package de.ubt.ai4.petter.recpro.recpro_evaluation_framework.util.api;

public final class ApiConstants {
    public static final String BASE = "/api/";

    public static final String BPM = BASE + "bpm/";
    public static final String BPM_MODELING = BPM + "modeling/";
    public static final String BPM_PROCESS_MODELING = BPM_MODELING + "process/";
    public static final String BPM_ACTIVITY_MODELING = BPM_MODELING + "activity/";

    public static final String ERC_DATA = BASE + "data/";
    public static final String ERC_DATA_MODELING = ERC_DATA + "modeling/";
    public static final String ERC_DATA_META_ATTRIBUTE_MODELING = ERC_DATA_MODELING + "metaAttribute/";
    public static final String ERC_DATA_ACTIVITY_PARAMETER_MODELING = ERC_DATA_MODELING + "activityParameter/";
    public static final String ERC_DATA_ENVIRONMENT_PARAMETER_MODELING = ERC_DATA_MODELING + "environmentParameter/";

    public static final String ERC_DATA_EXECUTION = ERC_DATA + "execution/";
    public static final String ERC_DATA_ACTIVITY_PARAMETER_EXECUTION = ERC_DATA_EXECUTION + "activityParameter/";


    public static final String RECPRO = BASE + "recpro/";
    public static final String RECPRO_MODELING = RECPRO + "modeling/";
    public static final String RECPRO_MODELING_ERC_DATA = RECPRO_MODELING + "erc/";
    public static final String RECPRO_MODELING_ERC_DATA_META_ATTRIBUTE_CONFIGURATION = RECPRO_MODELING_ERC_DATA + "metaAttributeConfiguration/";
    public static final String RECPRO_MODELING_ERC_DATA_ACTIVITY_PARAMETER_CONFIGURATION = RECPRO_MODELING_ERC_DATA + "activityParameterConfiguration/";
    public static final String RECPRO_MODELING_ERC_DATA_ENVIRONMENTAL_PARAMETER_CONFIGURATION = RECPRO_MODELING_ERC_DATA + "environmentalParameterConfiguration/";

    public static final String BPM_EXECUTION = BPM + "execution/";
    public static final String BPM_PROCESS_EXECUTION = BPM_EXECUTION + "process/";
    public static final String BPM_TASKLIST_EXECUTION = BPM_EXECUTION + "tasklist/";
    public static final String BPM_TASK_EXECUTION = BPM_EXECUTION + "task/";

    public static final String FEEDBACK = BASE + "feedback/";
    public static final String FEEDBACK_MODELING = FEEDBACK + "modeling/";
    public static final String FEEDBACK_EXECUTION = FEEDBACK + "execution/";

    public static final String RECOMMENDATION = BASE + "recommendation/";
    public static final String RECOMMENDATION_MODELING = RECOMMENDATION + "modeling/";
    public static final String RECOMMENDATION_EXECUTION = RECOMMENDATION + "execution/";

    public static final String RECOMMENDATION_DIMENSION = RECOMMENDATION + "dimension/";
    public static final String RECOMMENDATION_DIMENSION_MODELING = RECOMMENDATION_DIMENSION + "modeling/";
    public static final String RECOMMENDATION_DIMENSION_EXECUTION = RECOMMENDATION_DIMENSION + "execution/";

    public static final String FILTER = BASE + "filter/";
    public static final String FILTER_MODELING = FILTER + "modeling/";
    public static final String FILTER_EXECUTION = FILTER + "execution/";

    public static final String USER = BASE + "user/";
    public static final String USER_MODELING = USER + "modeling/";
    public static final String USER_EXECUTION = USER + "execution/";

    public static final String ID = "{id}";
    public static final String IDS = "{ids}";
    public static final String MAIL = "{mail}";
    public static final String LABEL = "{label}";

    public static final String GET = "get/";
    public static final String GET_ALL = "getAll";
    public static final String GET_ALL_IDS = "getAll/";
    public static final String GET_BY_IDS = "getByIds";
    public static final String GET_ACTIVE = "getActive/";
    public static final String GET_LATEST = "getLatest";
    public static final String GET_MODEL = "getModel/";
    public static final String GET_BY_MAIL = "getByMail/";
    public static final String GET_BY_LABEL = "getByLabel/";
    public static final String CREATE = "create";
    public static final String CREATE_ALL = "createAll";
    public static final String DELETE = "delete";
    public static final String UPDATE = "update";
    public static final String DELETE_ALL = "deleteAll";
    public static final String DELETE_ID = "delete/";
    public static final String START = "start/";
    public static final String SET_ATTRIBUTES = "setAttributes/";
    public static final String GET_ATTRIBUTES = "getAttributes/";
    public static final String CLAIM = "claim/";
    public static final String UNCLAIM = "unclaim/";
    public static final String ASSIGN = "assign/";
    public static final String INITIALIZE = "initialize";
    public static final String COMPLETE = "complete/";
    public static final String PRIORITY = "priority/";
    public static final String SET = "set/";
    public static final String GET_BY_PROCESS_ID = "getByProcessId/";
    public static final String GET_BY_ACTIVITY_ID = "getByActivityId/";
    public static final String GET_BY_META_ATTRIBUTE_ID = "getByMetaAttributeId/";
    public static final String GET_BY_ACTIVITY_PARAMETER_ID = "getByActivityParameterId/";
    public static final String GET_BY_TASK_INSTANCE_ID = "getByTaskInstanceId/";

    public static final String ASSIGNEE_ID = "assigneeId/";

    public static final String GET_DEFAULT = "getDefault";
    public static final String SET_DEFAULT = "setDefault/";
    public static final String SET_PRIORITY = "setPriority/";
    public static final String GET_BY_TYPE = "getByType/";

    public static final String ACTIVITY_ID = "activityId";
    public static final String META_ATTRIBUTE_ID = "metaAttributeId";
    public static final String TYPE = "type/";
    public static final String TYPE_ID = "{type}";

    public static final String USER_ID = "X-User-ID";
    public static final String APPLY_RECOMMENDATION = "applyRecommendation";
}

File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/util/ApiConstants.ts
export class ApiConstants {
  static BASE_API_URL = '';

  static initialize(baseUrl: string): void {
    if (!baseUrl) {
      throw new Error('BASE_API_URL is required for ApiConstants initialization.');
    }
    this.BASE_API_URL = `${baseUrl}/api/`;

    this.BPM = `${this.BASE_API_URL}bpm/`;
    this.BPM_MODELING = `${this.BPM}modeling/`;
    this.BPM_PROCESS_MODELING = `${this.BPM_MODELING}process/`;
    this.BPM_ACTIVITY_MODELING = `${this.BPM_MODELING}activity/`;

    this.BPM_EXECUTION = `${this.BPM}execution/`;
    this.BPM_PROCESS_EXECUTION = `${this.BPM_EXECUTION}process/`;
    this.BPM_TASK_EXECUTION = `${this.BPM_EXECUTION}task/`;
    this.BPM_TASKLIST_EXECUTION = `${this.BPM_EXECUTION}tasklist/`;


    this.FEEDBACK = `${this.BASE_API_URL}feedback/`;
    this.FEEDBACK_MODELING = `${this.FEEDBACK}modeling/`;
    this.FEEDBACK_EXECUTION = `${this.FEEDBACK}execution/`;

    this.ERC_DATA = `${this.BASE_API_URL}data/`;
    this.ERC_DATA_MODELING = `${this.ERC_DATA}modeling/`;
    this.ERC_DATA_META_ATTRIBUTE_MODELING = `${this.ERC_DATA_MODELING}metaAttribute/`;
    this.ERC_DATA_ACTIVITY_PARAMETER_MODELING = `${this.ERC_DATA_MODELING}activityParameter/`;
    this.ERC_DATA_ENVIRONMENT_PARAMETER_MODELING = `${this.ERC_DATA_MODELING}environmentParameter/`;

    this.ERC_DATA_EXECUTION= `${this.ERC_DATA}execution/`;
    this.ERC_DATA_ACTIVITY_PARAMETER_EXECUTION = `${this.ERC_DATA_EXECUTION}activityParameter/`;

    this.RECPRO= `${this.BASE_API_URL}recpro/`;
    this.RECPRO_MODELING = `${this.RECPRO}modeling/`;
    this.RECPRO_MODELING_ERC_DATA = `${this.RECPRO_MODELING}erc/`;
    this.RECPRO_MODELING_ERC_DATA_META_ATTRIBUTE_CONFIGURATION = `${this.RECPRO_MODELING_ERC_DATA}metaAttributeConfiguration/`;
    this.RECPRO_MODELING_ERC_DATA_ENVIRONMENTAL_PARAMETER_CONFIGURATION = `${this.RECPRO_MODELING_ERC_DATA}environmentalParameterConfiguration/`;
    this.RECPRO_MODELING_ERC_DATA_ACTIVITY_PARAMETER_CONFIGURATION = `${this.RECPRO_MODELING_ERC_DATA}activityParameterConfiguration/`;

    this.RECOMMENDATION= `${this.BASE_API_URL}recommendation/`;
    this.RECOMMENDATION_MODELING = `${this.RECOMMENDATION}modeling/`;
    this.RECOMMENDATION_EXECUTION = `${this.RECOMMENDATION}execution/`;

    this.RECOMMENDATION_DIMENSION = `${this.RECOMMENDATION}dimension/`;
    this.RECOMMENDATION_DIMENSION_MODELING = `${this.RECOMMENDATION_DIMENSION}modeling/`;
    this.RECOMMENDATION_DIMENSION_EXECUTION = `${this.RECOMMENDATION_DIMENSION}execution/`;

    this.USER = `${this.BASE_API_URL}user/`;
    this.USER_MODELING = `${this.USER}modeling/`;

  }



  // BPM Modeling:
  static BPM: string;
  static BPM_MODELING: string;
  static BPM_PROCESS_MODELING: string;
  static BPM_ACTIVITY_MODELING: string;

  // BPM Execution:
  static BPM_EXECUTION: string;
  static BPM_PROCESS_EXECUTION: string;
  static BPM_TASK_EXECUTION: string;
  static BPM_TASKLIST_EXECUTION: string;

  // FEEDBACK:
  static FEEDBACK: string;
  static FEEDBACK_MODELING: string;
  static FEEDBACK_EXECUTION: string;

  // ERC Data Modeling:
  static ERC_DATA: string;
  static ERC_DATA_MODELING: string;
  static ERC_DATA_META_ATTRIBUTE_MODELING: string;
  static ERC_DATA_ACTIVITY_PARAMETER_MODELING: string;
  static ERC_DATA_ENVIRONMENT_PARAMETER_MODELING: string;

  // ERC Data Execution:
  static ERC_DATA_EXECUTION: string;
  static ERC_DATA_ACTIVITY_PARAMETER_EXECUTION: string;

  // RECPRO:
  static RECPRO: string;
  static RECPRO_MODELING: string;
  static RECPRO_MODELING_ERC_DATA: string;
  static RECPRO_MODELING_ERC_DATA_META_ATTRIBUTE_CONFIGURATION: string;
  static RECPRO_MODELING_ERC_DATA_ENVIRONMENTAL_PARAMETER_CONFIGURATION: string;
  static RECPRO_MODELING_ERC_DATA_ACTIVITY_PARAMETER_CONFIGURATION: string;

  // Recommendation:
  static RECOMMENDATION: string;
  static RECOMMENDATION_MODELING: string;
  static RECOMMENDATION_EXECUTION: string;

  static RECOMMENDATION_DIMENSION: string;
  static RECOMMENDATION_DIMENSION_MODELING: string;
  static RECOMMENDATION_DIMENSION_EXECUTION: string;

  // User:
  static USER: string;
  static USER_MODELING: string;

  static readonly GET: string = 'get/';
  static readonly GET_BY_PROCESS_ID: string = 'getByProcessId/';
  static readonly GET_BY_ACTIVITY_ID: string = 'getByActivityId/';
  static readonly GET_BY_META_ATTRIBUTE_ID: string = 'getByMetaAttributeId/';
  static readonly GET_BY_ACTIVITY_PARAMETER_ID: string = 'getByActivityParameterId/';
  static readonly GET_BY_IDS: string = 'getByIds/';
  static readonly GET_ACTIVE: string = 'getActive/';
  static readonly SET_ACTIVE: string = 'setActive/';
  static readonly GET_ALL: string = 'getAll';
  static readonly CREATE: string = 'create';
  static readonly CREATE_ALL: string = 'createAll';
  static readonly UPDATE: string = 'update';
  static readonly DELETE: string = 'delete';
  static readonly DELETE_ID: string = 'delete/';
  static readonly INITIALIZE: string = 'initialize';

  static readonly START_ID = 'start/';
  static readonly CLAIM_ID: string = 'claim/';
  static readonly ASSIGN_ID: string = 'assign/';
  static readonly COMPLETE_ID: string = 'complete/';
  static readonly SET_PRIORITY: string = 'setPriority/';

  static readonly ACTIVITY_ID: string = 'activityId';
  static readonly META_ATTRIBUTE_ID: string = 'metaAttributeId';
  static readonly ASSIGNEE_ID: string = 'assigneeId/';
}
